src/components/Semester.vue

```diff
@@ -89,34 +116,33 @@ export default {
         this.isAddingNewModule = false;
       },
     },
-    isAddingNewModule(newValue) {
+    isAddingNewModule(newValue: boolean) {
+      const addModuleInput = ref<HTMLInputElement | null>(null);
+      if (!addModuleInput.value) {
+        return;
+      }
       if (newValue === false) {
-        this.$refs.addModuleInput.value = null;
+        addModuleInput.value?.setAttribute('value', '');
       } else {
         this.$nextTick(() => {
-          this.$refs.addModuleInput.focus();
+          addModuleInput.value?.focus();
         });
       }
     },
   },
-  data() {
-    return {
-      isAddingNewModule: false,
-    };
-  },
   methods: {
-    addModule(moduleName) {
-      this.$emit('on-add-module', moduleName, this.number);
+    addModule(event: Event) {
+      this.$emit('on-add-module', (<HTMLInputElement>event.currentTarget).value, this.number);
     },
     removeSemester() {
       this.$emit('on-remove-semester', this.number);
     },
-    countTotalEcts() {
+    countTotalEcts(): number {
       return this.modules.reduce((previousValue, module) => previousValue + module.ects, 0);
     },
     onDropEnd() {
-      this.$parent.updateUrlFragment();
+      this.$emit('on-drop-end');
```

Comment:
Yes, drag-n-drop still works 👌
I'm not sure, why we only went with `$parent`, while already using `$emit` everywhere else.

Reply:
Yes, drag-n-drop still works 👌
I'm not sure, why we only went with `$parent`, while already using `$emit` everywhere else.

Reply:
Yes. Thanks to `@on-drop-end="updateUrlFragment"` in the parent (`Home.vue`).
I also prefer this way of doing it, since it allows the parent to decide how to react.